src/main/java/com/aws/greengrass/componentmanager/ClientConfigurationUtils.java

```diff
@@ -51,8 +55,36 @@ public static String getGreengrassServiceEndpoint(DeviceConfiguration deviceConf
             logger.atError().setCause(e).log("Caught exception while parsing Nucleus args");
             throw new RuntimeException(e);
         }
-        return RegionUtils.getGreengrassDataPlaneEndpoint(Coerce.toString(deviceConfiguration.getAWSRegion()), stage,
-                Coerce.toInt(deviceConfiguration.getGreengrassDataPlanePort()));
+        // Use customer configured GG endpoint if it is set
+        String endpoint = Coerce.toString(deviceConfiguration.getGGDataEndpoint());
+        if (Utils.isEmpty(endpoint)) {
+            // Use customer configured IoT data endpoint if it is set
+            endpoint = Coerce.toString(deviceConfiguration.getIotDataEndpoint());
+        }
```

Comment:
IoT endpoint can't be empty since we validate that - so it would never fall back to the RegionUtils.

Do we want to use RegionUtils as default and use ggDataEndpoint as override if present?

Would this affect any firewall config if customers have setup a rule for greengrass-ats:8443 vs the iot data endpoint at 8443?